sim/deathknight/talents_blood.go

```diff
@@ -50,8 +50,27 @@ func (dk *Deathknight) ApplyBloodTalents() {
 	// Improved Rune Tap
 	// TODO: Implemented outside
 
-	// Spell Deflection
-	// TODO: Implement
+	// Spell Deflection && WoTN
+	dk.applyWillOfTheNecropolis()
+	dk.DynamicDamageTakenModifier = func(sim *core.Simulation, spellEffect *core.SpellEffect) {
+		if dk.Talents.SpellDeflection > 0 && spellEffect.ProcMask.Matches(core.ProcMaskSpellDamage) {
+			procChance := dk.GetStat(stats.Parry) / core.ParryRatingPerParryChance
+			dmgMult := 1.0 - 0.15*float64(dk.Talents.SpellDeflection)
+			if -1 < procChance {
+				spellEffect.Damage *= dmgMult
+			}
+		}
+
+		if dk.Talents.WillOfTheNecropolis > 0 {
+			if (dk.CurrentHealth()-spellEffect.Damage)/dk.MaxHealth() <= 0.35 {
+				spellEffect.Damage *= 0.85
+				if (dk.CurrentHealth()-spellEffect.Damage)/dk.MaxHealth() <= 0.35 {
+					dk.WillOfTheNecropolis.Activate(sim)
+					return
+				}
+			}
+		}
+	}
```

Comment:
The 'right' way to do that is probably just make this into a real aura callback, or at least make it a list of callbacks instead of just 1. Without using auras its going to prevent any aura metrics from showing, but we can go with this for now and see how it plays out, since this is only for a very small number of effects.